### PR TITLE
fix(discover) Honor discover2.tags_facet_enable_sampling when applying sampling  

### PR DESCRIPTION
**Pull request by fpacifici**
Depends on https://github.com/getsentry/sentry/pull/19882

The transactions table only included the sampling key clause in april (for on prem). There is no possible migration there without migrating to a new table. People may have already created that table and there is no support for data migrations in snuba yet.
If we start having users using tracing/preformance on prem, who created the table long before that they may run into troubles when they have enough data in the transactions table to trigger the sampling rate.
Snuba would reject those queries.

This honors the flag introduced in https://github.com/getsentry/sentry/pull/19882

### CODE REVIEW

**Review by BYK — APPROVED** [review]

File: src/sentry/web/forms/accounts.py [hunk 1]
```diff
@@ -321,442 +318,6 @@ def clean_password(self):
...
-        else:
-            UserOption.objects.set_value(
-                user=self.user,
-                key="workflow:notifications",
-                value=workflow_notifications_value,
-                project=None,
-            )
-
-
-class ProjectEmailOptionsForm(forms.Form):
```
Comment (markstory) [inline src/sentry/web/forms/accounts.py:692]: Why did these forms get removed if you're enabling a faceting option?

**Comment by fpacifici** [comment]
I did not try to remove forms,  @BYK  that was not part of the changeset when this was based on top of the PR that registered the option

File: src/sentry/web/forms/accounts.py (hunk 1 above)
Comment (fpacifici) [reply src/sentry/web/forms/accounts.py:692]: I don't see anything being removed in the commit 
https://github.com/getsentry/sentry/commit/691dd67778e14751e109345fea03a985502b6711

**Comment by BYK** [comment]
I think this is a GitHub mess up. They are having some issues right now. All I did was merging master and pushing, which was a clean merge.

**Pull request merged**